### PR TITLE
migrate to ts

### PR DESCRIPTION
**Pull request by itajaja**
this was more like an exercise to test cli and migrating to ts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,37 +2,83 @@
   "name": "@4c/env",
   "version": "1.2.0",
   "description": "Environment variable management and helpers",
-  "main": "env.js",
   "license": "MIT",
   "scripts": {
+    "build:es": "babel src -d es --env-name esm --ignore **/__tests__ --delete-dir-on-start -x .ts",
```
Comment (taion) [inline package.json]: es build seems unnecessary here

Comment (itajaja) [reply package.json]: forgive my ignorance, what's the purpose of the ES build at all?

**Comment by itajaja** [comment]
@jquense 	do you know what could go wrong with eslint?

**Comment by jquense** [comment]
sounds likes a package version mismatch if it's complaining about arguments...i'm not sure what specifically tho. Maybe our airbnb is mismatched with the local import-plugin?

File: .eslintrc [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
+      "node": true
+  },
   "rules": {
-    "prettier/prettier": "error"
-  }
+    "prettier/prettier": "error",
+    "global-require": "off",
+    "import/no-dynamic-require": "off"
+  },
+  "overrides": [
```
Comment (jquense) [inline .eslintrc]: lets remove this, and add 4catalyzer-jest to the extends above, i'll update `cli` for this as well

File: src/env.ts [hunk 3]
```diff
@@ -22,27 +22,27 @@ module.exports = {
...
 
-  unset(key) {
+  unset(key: string) {
     delete process.env[key];
   },
 
   loadFromFile,
 
-  load(env, { force = true } = {}) {
+  load(env: string | undefined, { force = true } = {}) {
```
[the remark this answers is not in the document]
Comment (itajaja) [reply src/env.ts]: uh. dunno.

File: package.json (hunk 1 above)
Comment (taion) [reply package.json]: it allows scope hoisting in webpack

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    "noImplicitAny": true,
+    "noImplicitThis": true,
+    "noUnusedParameters": true,
+    "skipLibCheck": true,
+    "sourceMap": true,
+    "strict": true,
+    "strictFunctionTypes": false,
+    "target": "esnext"
+  },
+  "include": ["src"]
```
[the remark this answers is not in the document]
Comment (itajaja) [reply tsconfig.json]: this value will not from `4c/tsconfig`.

[the remark this answers is not in the document]
Comment (jquense) [reply tsconfig.json]: I mean in general for the compiler options

[the remark this answers is not in the document]
Comment (itajaja) [reply tsconfig.json]: yep doing it

**Comment by itajaja** [comment]
all gtg.

File: .eslintrc [hunk 5]
```diff
@@ -12,18 +12,7 @@
...
-        "no-await-in-loop": "off",
-        "no-console": "off",
-        "import/no-dynamic-require": "off",
-        "jest/no-disabled-tests": "warn",
-        "jest/no-focused-tests": "error",
-        "jest/no-identical-title": "error",
-        "jest/prefer-to-have-length": "warn",
-        "jest/valid-expect": "error"
-      }
+      "plugins": ["jest", "eslint-config-4catalyzer-jest"]
```
Comment (jquense) [inline .eslintrc]: wrong place, should go in the top level `extends`, see my changes in cli

Comment (itajaja) [reply .eslintrc]: actually I have a question about [it](https://github.com/4Catalyzer/cli/pull/19#discussion_r249580091)

**Pull request merged**